src/pages/smart-contracts/setting-up-local-environment.md

```diff
@@ -0,0 +1,332 @@
+- [Todo's web with Connect](https://github.com/blockstack/blockstack-todos/)
+
+## Step 1: Set up Stacks Node and Blockchain API
+
+The easiest way to spin up both of these locally is by booting up a docker image:
+
+```bash
+docker run -p 3999:3999 blockstack/stacks-blockchain-api-standalone mocknet
+```
+
+The initial start will take a few minutes. As soon as it is complete, you should be able to open up `http://localhost:3999/` and see:
+
+```js
+{"status":"ready"}
+```
+
+## Step 2: Set up Stacks Explorer
+
+The Stacks Explorer makes it easy way to review transactions. Let's spin up a local version:
+
+```bash
+git clone https://github.com/blockstack/explorer.git
+cd explorer
+yarn
+yarn dev
+```
+
+You should now be able to open up `http://localhost:3000/` and see the homepage. **Make sure to select `Localhost` from the network list on the top right side of the screen!**
+
+Now, click on "Recent transactions" and you should see the two contracts.
+
+## Step 3: Create sample contract project
+
+Next, let's create a sample contract project:
+
+```bash
+cd ..
+mkdir contracts
+cd contracts
+npm init clarity-starter
+```
+
+Select "Counter" and wait until all dependencies are installed.
+
+Upon completion, open up the folder using [VSCode](https://code.visualstudio.com/). This editor is recommended because there are extension available to make Clarity development easier. Please install these extensions:
+
+- [Clarity](https://marketplace.visualstudio.com/items?itemName=blockstack.clarity), the official language extension by Blockstack that defines the Clarity language for VS Code and provides auto-complete and syntax highlighting.
+- [clarity-lsp](https://marketplace.visualstudio.com/items?itemName=lgalabru.clarity-lsp), which adds inline help functionality for Clarity to VS Code
+- [Rainbow Brackets](https://marketplace.visualstudio.com/items?itemName=2gua.rainbow-brackets), which adds helpful colorization of matching pairs of parentheses while you code
+
+!> Do not just search for "Clarity" in the marketplace. You might install the wrong extensions by accident.
+
+## Step 4: Create contract deployment script
+
+When you develop your contract, you will often redeploy your `.clar` files. The fastest way is to create a deployment script.
+
+In the same contract project folder as before, install dependencies:
+
+```bash
+npm i @stacks/network @stacks/transactions --saveDev
+```
+
+Next, create the deployment script file:
+
+```bash
+mkdir scripts
+cd scripts
+touch deploy-contract.js
+```
+
+Finally, paste the following code:
+
+```js
+const readline = require('readline');
+const fs = require('fs');
+const { makeContractDeploy, broadcastTransaction } = require('@stacks/transactions');
+const { StacksTestnet } = require('@stacks/network');
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+// set to local mocknet
+const network = new StacksTestnet();
+network.coreApiUrl = 'http://localhost:3999';
```

Comment:
This can be changed to `const network = new StacksMocknet();` as soon as stacks.js monorepo is released and dependency updated.